### ISSUE TITLE
Responsive stop

### ISSUE THREAD

**Issue by brian-silverman** [scottsilverlabs/raspberrystem-ide#53]
Need stop thread to be higher priority so infinite loops are still responsive

**Comment by scottsilverlabs** [comment]
Stef - when you get a chance, can you look at this one?  I'm hoping you'll have a chance to look at this by early next week.  If not, I will.  Its the only remaining issue that I'd like to get resolved before July 6th (when our first camp starts).

**Comment by brian-silverman** [comment]
Stef - FYI, I've reverted your fixes on this because there was only marginal increase in performance and they broke error highlighting (and sometimes cause the top line of a Traceback to get cutoff).  Instead, I've added a quickfix to just limit the lines of the buffer to the last 100 printed (I need these fixes for Monday).  Note: my fix only also has a so-so performance increase when run on the Pi, but it doesn't break error highlighting.

The main issue seems to be in the append - as the buffer grows, the appends time (which requires a copy) grows longer until the javascript grinds to a halt.

I think a reasonable fix would be to:
* Have the go server maintain a buffer of max ~2MB (probably similar to the second fix you had that I reverted)
* When the running python program has a lot of output, send only the last N lines that will fill the output window.  For example, when sending output, if that last send was > 0.1 seconds ago, just send immediately.  But if the last send was < 0.1 seconds ago, just buffer it (and set a 0.1 second timer to send it later.

If that doesn't make sense, or if you have other ideas, let me know...
